From ecae842ee60ae4c1d78c1de7d9c6d3238d069ab6 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Tue, 21 Sep 2021 12:08:25 +0200
Subject: [PATCH] Allow using loose requirements

Disabled by default. Useful for distribution packaging where we really
can't ship specific package versions depending on the package that needs
it. Distributions are responsible to make stuff works themselves anyway

The same is done in mycroft-core already
---
 setup.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/setup.py b/setup.py
index 4244cf4..74494cb 100644
--- a/setup.py
+++ b/setup.py
@@ -32,6 +32,9 @@ def required(requirements_file):
     base_dir = os.path.abspath(os.path.dirname(__file__))
     with open(os.path.join(base_dir, requirements_file), 'r') as f:
         requirements = f.read().splitlines()
+        if 'MYCROFT_LOOSE_REQUIREMENTS' in os.environ:
+            print('USING LOOSE REQUIREMENTS!')
+            requirements = [r.replace('==', '>=') for r in requirements]
         return [pkg for pkg in requirements
                 if pkg.strip() and not pkg.startswith("#")]
 
-- 
2.33.0

